homeassistant/components/media_player/anthemav.py

```diff
@@ -0,0 +1,179 @@
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA, SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_SELECT_SOURCE,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET, MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT, STATE_OFF, STATE_ON, STATE_UNKNOWN,
+    EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['anthemav==1.1.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'anthemav'
+
+DEFAULT_NAME = "Anthem AVR"
+DEFAULT_PORT = 14999
+
+SUPPORT_ANTHEMAV = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    })
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up our socket to the AVR."""
+    import anthemav
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+
+    _LOGGER.info('Provisioning Anthem AVR device at %s:%d', host, port)
+
+    def anthemav_update_callback(message):
+        """Receive notification from transport that new data exists."""
+        _LOGGER.info('Received update callback from AVR: %s', message)
+        hass.async_add_job(device.async_update_ha_state())
+
+    avr = yield from anthemav.Connection.create(
+        host=host, port=port, loop=hass.loop,
+        update_callback=anthemav_update_callback)
+
+    device = AnthemAVR(hass, avr)
+
+    _LOGGER.debug('dump_devicedata: '+device.dump_avrdata)
+    _LOGGER.debug('dump_conndata: '+avr.dump_conndata)
+    _LOGGER.debug('dump_rawdata: '+avr.protocol.dump_rawdata)
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, device.avr.close)
+    yield from async_add_devices([device])
+
+
+class AnthemAVR(MediaPlayerDevice):
+    """Entity reading values from Anthem AVR protocol."""
+
+    def __init__(self, hass, avr):
+        """"Initialize entity with hass, host, and port."""
+        super().__init__()
+        self.hass = hass
+        self.avr = avr
+
+    def _lookup(self, propname, dval):
+        return getattr(self.reader, propname, dval)
+
+    @property
+    def reader(self):
+        """Expose the protocol with smart wrapper."""
+        return self.avr.protocol
+
+    @property
+    def supported_media_commands(self):
+        """Return flag of media commands that are supported."""
+        return SUPPORT_ANTHEMAV
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return name of device."""
+        return self._lookup('model', DEFAULT_NAME)
+
+    @property
+    def state(self):
+        """Return state of power on/off."""
+        pwrstate = self._lookup('power', None)
```

Comment:
Just as a code style change we should set the default `dval` in `_lookup` to `None` and remove the argument here.